### PR TITLE
NAS-117680 / 22.12 / NAS-117680: update ix-icon for dataset details

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

1. Datasets > choose a dataset. This should result in a change of an icon positioned "Details for" at the details section.
2. Refresh page, and choose another node of datasets tree. This should result in another change of an icon.

### CODE REVIEW

File: src/app/pages/datasets/components/dataset-icon/dataset-icon.component.html [hunk 1]
```diff
@@ -1,3 +1,4 @@
 <!-- TODO: https://jira.ixsystems.com/browse/NAS-116793 -->
 <!-- TODO: Remove me -->
-<ix-icon [name]="name"></ix-icon>
+<ix-icon *ngIf="isRoot" name="ix:dataset"></ix-icon>
```
Comment (undsoft) [inline src/app/pages/datasets/components/dataset-icon/dataset-icon.component.html:3]: Feels like this should be within `name` getter.

File: src/app/modules/ix-icon/ix-icon.component.ts [hunk 2]
```diff
@@ -70,15 +70,24 @@ export class IxIconComponent extends MatIcon implements OnInit {
...
-    if (this.iconName.startsWith('ix:')) {
-      this.svgIcon = this.iconName;
-    } else if (this.iconName.startsWith('mdi')) {
+    this.updateIcon(this.iconName);
+    super.ngOnInit();
+  }
+
+  ngOnChanges(): void {
+    this.updateIcon(this.iconName);
+  }
```
Comment (undsoft) [inline src/app/modules/ix-icon/ix-icon.component.ts:79]: Should parent `onChanges` be called as well?

**Review by undsoft — APPROVED** [review]
Icons change well, don't see any issues.

File: src/app/modules/ix-icon/ix-icon.component.ts (hunk 2 above)
Comment (bvasilenko) [reply src/app/modules/ix-icon/ix-icon.component.ts:79]: Parent class has no `ngOnChanges` method.

**Pull request merged**